### PR TITLE
Make the linera-net-tests run on `remote-net` instead of `storage-service`.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

It is a long running objective to have the `linera-net-tests` running on a single set of validators.

## Proposal

The following was done:
* The `storage-service` tests are disabled in `linera_net_tests.rs` by renaming their title from `storage_service_grpc` to `storage_test_service_grpc`.
* The related work for making the `remote-net` are added to the CI file.

## Test Plan

(A) The CI should pass. The change is that some tests pass from running on the `storage-service` to running on the `remote-net`.

(B) The runtime of the 20 tests of linera_net_tests.rs is 691 seconds on the `storage-service` and 345 seconds on the `remote-net` by using 1 validator and 1 shard. However, when running on 4 validators and 4 shards, the gains are much more modest with 638 seconds. Due to the additional compilation work this translates into an increase of the CI runtime.

## Release Plan

None of the changes affect the `testnet` and `devnet`.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -60,22 +60,33 @@ jobs:
...
+        echo "LINERA_FAUCET_URL=http://localhost:8079" >> "$GITHUB_ENV"
     - name: Build example applications
       run: |
         cd examples
         cargo build --locked --release --target wasm32-unknown-unknown
-    - name: Run end-to-end tests
+    - name: Run the storage-service instance and the storage-service tests
       run: |
         cargo run --release -p linera-storage-service -- memory --endpoint $LINERA_STORAGE_SERVICE &
         cargo test --features storage-service,unstable-oracles -- storage_service --nocapture
```
Comment (afck) [inline .github/workflows/rust.yml:73]: Does this line still do anything?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -60,22 +60,33 @@ jobs:
...
+        cargo build --features storage-service
+        mkdir /tmp/WORK
+        cargo run --features storage-service --bin linera -- net up --storage-config-namespace service:tcp:localhost:1235:table --policy-config devnet --path /tmp/WORK &
     - name: Compile Wasm test modules for Witty integration tests
       run: |
         cargo build -p linera-witty-test-modules --target wasm32-unknown-unknown
+    - name: Run the remote-net tests
+      run: |
+        cargo run --bin linera -- faucet --amount 1000 --port 8079 &
+        cargo test -p linera-service remote_net_grpc --features remote-net
```
Comment (afck) [inline .github/workflows/rust.yml:89]: We should probably run this with the `unstable-oracles` feature.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (MathieuDutSik) [reply .github/workflows/rust.yml:73]: Yes, it does:
* The `README` tests.
* The tests in `local_net_tests.rs`
* 25 tests in `client::client_tests::*::storage_service`.
* 1 test `test_end_to_end_operations_indexer::storage_service_grpc`.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (MathieuDutSik) [reply .github/workflows/rust.yml:89]: No, because this creates the error ```
error: none of the selected packages contains these features: unstable-oracles
```

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -79,7 +79,7 @@ jobs:
       run: |
         cargo build --features storage-service
         mkdir /tmp/WORK
-        cargo run --features storage-service --bin linera -- net up --storage-config-namespace service:tcp:localhost:1235:table --policy-config devnet --path /tmp/WORK &
+        cargo run --features storage-service --bin linera -- net up --storage-config-namespace service:tcp:localhost:1235:table --policy-config devnet --path /tmp/WORK --validators 4 --shards 4 &
```
Comment (ma2bd) [inline .github/workflows/rust.yml:82]: Is `service:tcp:localhost:1235:table` supposed to be a namespace?
How can the validators share the same namespace and the same service?

Comment (MathieuDutSik) [reply .github/workflows/rust.yml:82]: The `service:tcp:localhost:1235:table` is parsed into the following:
* `service` means that `storage-service` is being used.
* `tcp` is mandated for the protocol being used.
* `localhost` is the host being used.
* `1235` is the port being used for the `storage-service`.

Looking in an inner way, what is going on we find that `linera net up` is using the `handle_net_up_service` which itself calls the `LocalNetConfig`. Therefore, the system being used is the one  implemented for that instantiaton of the trait `LineraNetConfig`.

The implementation of the trait is in `local_net.rs` which calls the `fn initialize_storage`. The namespace being used are then specific to the validators. The key line to see is 
```
        let namespace = format!("{}_server_{}_db", self.namespace, validator);
```
In other words, the namespace being used are `table_server_X_db` with X=0, 1, 2, 3 in that case.

In summary, each validator has its namespace which are shared by all the shards of a given validator. That design is the one that we have used since about May 2023.

Comment (ma2bd) [reply .github/workflows/rust.yml:82]: I was mostly commenting on the name of the option `--storage-config-namespace`

Comment (MathieuDutSik) [reply .github/workflows/rust.yml:82]: Ok, I can rename it as `storage`.

Comment (ma2bd) [reply .github/workflows/rust.yml:82]: --storage-config also possible

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -60,22 +60,37 @@ jobs:
...
         cargo test -p linera-ethereum --features ethereum
         cargo test test_wasm_end_to_end_ethereum_tracker --features ethereum,storage-service,unstable-oracles
+    - name: Run the benchmark test
+      run: |
+        cargo build --locked -p linera-service --bin linera-benchmark --features benchmark,storage-service
+        cargo test --locked -p linera-service --features benchmark,storage-service benchmark
+    - name: Run the validators
+      run: |
+        cargo build --features storage-service
+        mkdir /tmp/WORK
```
Comment (afck) [inline .github/workflows/rust.yml:84]: Maybe `/tmp/local-linera-net`?

File: CLI.md [hunk 5]
```diff
@@ -869,7 +869,7 @@ Start a Local Linera Network
 
 * `--testing-prng-seed <TESTING_PRNG_SEED>` — Force this wallet to generate keys using a PRNG and a given seed. USE FOR TESTING ONLY
 * `--path <PATH>` — Run with a specific path where the wallet and validator input files are. If none, then a temporary directory is created
-* `--storage-config-namespace <STORAGE_CONFIG_NAMESPACE>` — Run with a specific storage. If none, then a linera-storage-service is spanned on a random free port
+* `--storage <STORAGE>` — Run with a specific storage. If none, then a linera-storage-service is spanned on a random free port
```
Comment (afck) [inline CLI.md:872]: Should this say "spawned"? Or maybe "started" or "run"?

**Review by afck — APPROVED** [review]

**Review by jvff — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 4 above)
Comment (MathieuDutSik) [reply .github/workflows/rust.yml:84]: Ok.

File: CLI.md (hunk 5 above)
Comment (MathieuDutSik) [reply CLI.md]: Corrected.
